### ISSUE TITLE
Error when building with node v18.20.x

### ISSUE THREAD

**Issue by benz0li** [microsoft/node-native-keymap#58]
command:

```bash
mkdir -p ~/projects/microsoft \
  && git clone https://github.com/microsoft/node-native-keymap.git \
    ~/projects/microsoft/node-native-keymap \
  && cd ~/projects/microsoft/node-native-keymap \
  && sudo npm install -g node-gyp \
  && node-gyp configure -d \
  && node-gyp build
```

output:

```
Cloning into '/home/jovyan/projects/microsoft/node-native-keymap'...
remote: Enumerating objects: 986, done.
remote: Counting objects: 100% (103/103), done.
remote: Compressing objects: 100% (38/38), done.
remote: Total 986 (delta 71), reused 85 (delta 63), pack-reused 883
Receiving objects: 100% (986/986), 279.75 KiB | 3.63 MiB/s, done.
Resolving deltas: 100% (584/584), done.

added 101 packages in 5s

14 packages are looking for funding
  run `npm fund` for details
npm notice 
npm notice New patch version of npm available! 10.5.0 -> 10.5.2
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.5.2
npm notice Run npm install -g npm@10.5.2 to update!
npm notice 
gyp info it worked if it ends with ok
gyp info using node-gyp@10.1.0
gyp info using node@18.20.2 | linux | x64
gyp info find Python using Python version 3.12.3 found at "/usr/local/bin/python3"

gyp http GET https://nodejs.org/download/release/v18.20.2/node-v18.20.2-headers.tar.gz
gyp http 200 https://nodejs.org/download/release/v18.20.2/node-v18.20.2-headers.tar.gz
gyp http GET https://nodejs.org/download/release/v18.20.2/SHASUMS256.txt
gyp http 200 https://nodejs.org/download/release/v18.20.2/SHASUMS256.txt
gyp info spawn /usr/local/bin/python3
gyp info spawn args [
gyp info spawn args '/usr/local/lib/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args 'binding.gyp',
gyp info spawn args '-f',
gyp info spawn args 'make',
gyp info spawn args '-I',
gyp info spawn args '/home/jovyan/projects/microsoft/node-native-keymap/build/config.gypi',
gyp info spawn args '-I',
gyp info spawn args '/usr/local/lib/node_modules/node-gyp/addon.gypi',
gyp info spawn args '-I',
gyp info spawn args '/home/jovyan/.cache/node-gyp/18.20.2/include/node/common.gypi',
gyp info spawn args '-Dlibrary=shared_library',
gyp info spawn args '-Dvisibility=default',
gyp info spawn args '-Dnode_root_dir=/home/jovyan/.cache/node-gyp/18.20.2',
gyp info spawn args '-Dnode_gyp_dir=/usr/local/lib/node_modules/node-gyp',
gyp info spawn args '-Dnode_lib_file=/home/jovyan/.cache/node-gyp/18.20.2/<(target_arch)/node.lib',
gyp info spawn args '-Dmodule_root_dir=/home/jovyan/projects/microsoft/node-native-keymap',
gyp info spawn args '-Dnode_engine=v8',
gyp info spawn args '--depth=.',
gyp info spawn args '--no-parallel',
gyp info spawn args '--generator-output',
gyp info spawn args 'build',
gyp info spawn args '-Goutput_dir=.'
gyp info spawn args ]
gyp info ok 
gyp info it worked if it ends with ok
gyp info using node-gyp@10.1.0
gyp info using node@18.20.2 | linux | x64
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Debug', '-C', 'build' ]
make: Entering directory '/home/jovyan/projects/microsoft/node-native-keymap/build'
  CXX(target) Debug/obj.target/keymapping/src/string_conversion.o
  CXX(target) Debug/obj.target/keymapping/src/keymapping.o
In file included from ../src/keymapping.cc:10:
../src/keymapping.cc: In function ‘napi_value__* vscode_keyboard::OnDidChangeKeyboardLayoutImpl(napi_env, napi_callback_info)’:
../src/keymapping.cc:100:50: error: invalid conversion from ‘void (*)(napi_env, void*, void*)’ {aka ‘void (*)(napi_env__*, void*, void*)’} to ‘node_api_nogc_finalize’ {aka ‘void (*)(const napi_env__*, void*, void*)’} [-fpermissive]
  100 |                                                  FinalizeThreadsafeFunction, NULL, NotifyJS,
      |                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                  |
      |                                                  void (*)(napi_env, void*, void*) {aka void (*)(napi_env__*, void*, void*)}
../src/common.h:64:10: note: in definition of macro ‘NAPI_CALL_BASE’
   64 |     if ((the_call) != napi_ok) {                                         \
      |          ^~~~~~~~
../src/keymapping.cc:99:3: note: in expansion of macro ‘NAPI_CALL’
   99 |   NAPI_CALL(env, napi_create_threadsafe_function(env, func, NULL, resource_name, 0, 1, NULL,
      |   ^~~~~~~~~
In file included from /home/jovyan/.cache/node-gyp/18.20.2/include/node/node.h:79,
                 from ../src/keymapping.cc:6:
/home/jovyan/.cache/node-gyp/18.20.2/include/node/node_api.h:212:56: note:   initializing argument 8 of ‘napi_status napi_create_threadsafe_function(napi_env, napi_value, napi_value, napi_value, size_t, size_t, void*, node_api_nogc_finalize, void*, napi_threadsafe_function_call_js, napi_threadsafe_function__**)’
  212 |                                 node_api_nogc_finalize thread_finalize_cb,
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
make: *** [keymapping.target.mk:114: Debug/obj.target/keymapping/src/keymapping.o] Error 1
make: Leaving directory '/home/jovyan/projects/microsoft/node-native-keymap/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack at ChildProcess.<anonymous> (/usr/local/lib/node_modules/node-gyp/lib/build.js:209:23)
gyp ERR! System Linux 6.1.0-20-amd64
gyp ERR! command "/usr/local/bin/node" "/usr/local/bin/node-gyp" "build"
gyp ERR! cwd /home/jovyan/projects/microsoft/node-native-keymap
gyp ERR! node -v v18.20.2
gyp ERR! node-gyp -v v10.1.0
gyp ERR! not ok
```

**Comment by benz0li** [comment]
The same command works fine when executed after

```bash
docker run --rm -u root -e GRANT_SUDO=1 -ti glcr.b-data.ch/jupyterlab/python/scipy:3.12.3-devtools bash
```

which provides node v18.19.1.

output:

```
Cloning into '/home/jovyan/projects/microsoft/node-native-keymap'...
remote: Enumerating objects: 986, done.
remote: Counting objects: 100% (58/58), done.
remote: Compressing objects: 100% (36/36), done.
remote: Total 986 (delta 26), reused 43 (delta 20), pack-reused 928
Receiving objects: 100% (986/986), 281.63 KiB | 9.08 MiB/s, done.
Resolving deltas: 100% (584/584), done.

added 101 packages in 4s

14 packages are looking for funding
  run `npm fund` for details
npm notice 
npm notice New minor version of npm available! 10.2.4 -> 10.5.2
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.5.2
npm notice Run npm install -g npm@10.5.2 to update!
npm notice 
gyp info it worked if it ends with ok
gyp info using node-gyp@10.1.0
gyp info using node@18.19.1 | linux | x64
gyp info find Python using Python version 3.12.3 found at "/usr/local/bin/python3"

gyp http GET https://nodejs.org/download/release/v18.19.1/node-v18.19.1-headers.tar.gz
gyp http 200 https://nodejs.org/download/release/v18.19.1/node-v18.19.1-headers.tar.gz
gyp http GET https://nodejs.org/download/release/v18.19.1/SHASUMS256.txt
gyp http 200 https://nodejs.org/download/release/v18.19.1/SHASUMS256.txt
gyp info spawn /usr/local/bin/python3
gyp info spawn args [
gyp info spawn args '/usr/local/lib/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args 'binding.gyp',
gyp info spawn args '-f',
gyp info spawn args 'make',
gyp info spawn args '-I',
gyp info spawn args '/home/jovyan/projects/microsoft/node-native-keymap/build/config.gypi',
gyp info spawn args '-I',
gyp info spawn args '/usr/local/lib/node_modules/node-gyp/addon.gypi',
gyp info spawn args '-I',
gyp info spawn args '/home/jovyan/.cache/node-gyp/18.19.1/include/node/common.gypi',
gyp info spawn args '-Dlibrary=shared_library',
gyp info spawn args '-Dvisibility=default',
gyp info spawn args '-Dnode_root_dir=/home/jovyan/.cache/node-gyp/18.19.1',
gyp info spawn args '-Dnode_gyp_dir=/usr/local/lib/node_modules/node-gyp',
gyp info spawn args '-Dnode_lib_file=/home/jovyan/.cache/node-gyp/18.19.1/<(target_arch)/node.lib',
gyp info spawn args '-Dmodule_root_dir=/home/jovyan/projects/microsoft/node-native-keymap',
gyp info spawn args '-Dnode_engine=v8',
gyp info spawn args '--depth=.',
gyp info spawn args '--no-parallel',
gyp info spawn args '--generator-output',
gyp info spawn args 'build',
gyp info spawn args '-Goutput_dir=.'
gyp info spawn args ]
gyp info ok 
gyp info it worked if it ends with ok
gyp info using node-gyp@10.1.0
gyp info using node@18.19.1 | linux | x64
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Debug', '-C', 'build' ]
make: Entering directory '/home/jovyan/projects/microsoft/node-native-keymap/build'
  CXX(target) Debug/obj.target/keymapping/src/string_conversion.o
  CXX(target) Debug/obj.target/keymapping/src/keymapping.o
  CXX(target) Debug/obj.target/keymapping/deps/chromium/x/keysym_to_unicode.o
  CXX(target) Debug/obj.target/keymapping/src/keyboard_x.o
  SOLINK_MODULE(target) Debug/obj.target/keymapping.node
  COPY Debug/keymapping.node
make: Leaving directory '/home/jovyan/projects/microsoft/node-native-keymap/build'
gyp info ok
```

**Comment by benz0li** [comment]
Either fix or update https://github.com/microsoft/vscode/wiki/How-to-Contribute#prerequisites so it says

> - [Node.JS](https://nodejs.org/en/), **x64**, version `>=18.15.x and <18.20`

Ping @deepak1556
